File: ruby/game.rb
require "./constants.rb"
require "./strings.rb"

class Game
  def print_keys
    puts STR_ROCK + ": " + ROCK.to_s
    puts STR_PAPER + ": " + PAPER.to_s
    puts STR_SCISSORS + ": " + SCISSORS.to_s
  end

  def handle_input(input)
    case input
    when ROCK
      return STR_ROCK
    when PAPER
      return STR_PAPER
    when SCISSORS
      return STR_SCISSORS
    else 
      return STR_INVALID
    end
  end

  def get_win_message(user_input, ai_input)
    if user_input == ai_input
      return STR_TIE
    elsif user_input == PAPER && ai_input == ROCK
      return STR_USER_WIN
    elsif user_input == PAPER && ai_input == SCISSORS
      return STR_AI_WIN
    elsif user_input == ROCK && ai_input == PAPER
      return STR_AI_WIN
    elsif user_input == ROCK && ai_input == SCISSORS
      return STR_USER_WIN
    elsif user_input == SCISSORS && ai_input == PAPER
      return STR_USER_WIN
    elsif user_input == SCISSORS && ai_input == ROCK
      return STR_AI_WIN
    else
      return STR_INVALID
    end
  end
end

File: python/Game.py
from Constants import ROCK, PAPER, SCISSORS
from Strings import STR_ROCK, STR_PAPER, STR_SCISSORS, STR_INVALID, STR_TIE, STR_AI_WIN, STR_USER_WIN

def print_keys():
  print(STR_ROCK + ": "+ str(ROCK))
  print(STR_PAPER + ": "+ str(PAPER))
  print(STR_SCISSORS + ": "+ str(SCISSORS))

def handle_input(input):
  match input:
    case int(ROCK):
      return STR_ROCK
    case int(PAPER):
      return STR_PAPER
    case int(SCISSORS):
      return STR_SCISSORS
    case _:
      return STR_INVALID

def get_win_message(user_input, ai_input):
  if user_input == ai_input:
    return STR_TIE
  elif user_input == PAPER and ai_input == ROCK:
    return STR_AI_WIN
  elif user_input == PAPER and ai_input == SCISSORS:
    return STR_USER_WIN
  elif user_input == ROCK and ai_input == PAPER:
    return STR_AI_WIN
  elif user_input == ROCK and ai_input == SCISSORS:
    return STR_USER_WIN
  elif user_input == SCISSORS and ai_input == PAPER:
    return STR_USER_WIN
  elif user_input == SCISSORS and ai_input == ROCK:
    return STR_AI_WIN
  else:
    return STR_INVALID